The job **`mingw-check`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/7617707027/job/20747307479) [(plain)](https://github.com/rust-lang/rust/commit/1f56852f2fce645ef79bd7ca4d0492ba69a73f89/checks/20747307479/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
GITHUB_ACTION=__run_7
GITHUB_ACTIONS=true
GITHUB_ACTION_REF=
GITHUB_ACTION_REPOSITORY=
GITHUB_ACTOR=chadnorvell
GITHUB_API_URL=https://api.github.com
GITHUB_BASE_REF=master
GITHUB_ENV=/home/runner/work/_temp/_runner_file_commands/set_env_352132ee-b69a-4da4-9b6c-96c4523ba849
GITHUB_EVENT_NAME=pull_request
---
GITHUB_SERVER_URL=https://github.com
GITHUB_SHA=928c03de58c15523f49c489e8584aacdd874678e
GITHUB_STATE=/home/runner/work/_temp/_runner_file_commands/save_state_352132ee-b69a-4da4-9b6c-96c4523ba849
GITHUB_STEP_SUMMARY=/home/runner/work/_temp/_runner_file_commands/step_summary_352132ee-b69a-4da4-9b6c-96c4523ba849
GITHUB_TRIGGERING_ACTOR=chadnorvell
GITHUB_WORKFLOW_REF=rust-lang/rust/.github/workflows/ci.yml@refs/pull/120250/merge
GITHUB_WORKFLOW_SHA=928c03de58c15523f49c489e8584aacdd874678e
GITHUB_WORKSPACE=/home/runner/work/rust/rust
GOROOT_1_19_X64=/opt/hostedtoolcache/go/1.19.13/x64
---
info: ES-Check: there were no ES version matching errors!  🎉
+ eslint -c ../src/librustdoc/html/static/.eslintrc.js ../src/librustdoc/html/static/js/externs.js ../src/librustdoc/html/static/js/main.js ../src/librustdoc/html/static/js/scrape-examples.js ../src/librustdoc/html/static/js/search.js ../src/librustdoc/html/static/js/settings.js ../src/librustdoc/html/static/js/src-script.js ../src/librustdoc/html/static/js/storage.js

/checkout/src/librustdoc/html/static/js/storage.js
  31:32  error  Unnecessary escape character: \(  no-useless-escape
  31:34  error  Unnecessary escape character: \)  no-useless-escape
  31:36  error  Unnecessary escape character: \[  no-useless-escape
  31:40  error  Unnecessary escape character: \{  no-useless-escape
  31:42  error  Unnecessary escape character: \}  no-useless-escape
  31:44  error  Unnecessary escape character: \*  no-useless-escape
  31:46  error  Unnecessary escape character: \"  no-useless-escape
  31:48  error  Unnecessary escape character: \'  no-useless-escape
  31:50  error  Unnecessary escape character: \`  no-useless-escape

✖ 9 problems (9 errors, 0 warnings)
  local time: Mon Jan 22 22:02:53 UTC 2024
  network time: Mon, 22 Jan 2024 22:02:53 GMT
##[error]Process completed with exit code 1.
Post job cleanup.
```

</details>

--- a/src/librustdoc/html/static/js/storage.js
+++ b/src/librustdoc/html/static/js/storage.js
@@ -101,6 +101,14 @@ const getVar = (function getVar(name) {
 });
 
 function switchTheme(newThemeName, saveTheme) {
+    const themeNames = getVar("themes").split(",").filter(t => t);
+    themeNames.push(...builtinThemes);
+
+    // Ensure that the new theme name is among the defined themes
+    if (themeNames.indexOf(newThemeName) === -1) {
+        return;
+    }
+
     // If this new value comes from a system setting or from the previously
     // saved theme, no need to save it.
     if (saveTheme) {
@@ -115,7 +123,7 @@ function switchTheme(newThemeName, saveTheme) {
             window.currentTheme = null;
         }
     } else {
-        const newHref = getVar("root-path") + newThemeName +
+        const newHref = getVar("root-path") + encodeURIComponent(newThemeName) +
             getVar("resource-suffix") + ".css";
         if (!window.currentTheme) {
             // If we're in the middle of loading, document.write blocks

